code/game/objects/structures/headpike.dm

```diff
@@ -33,9 +33,9 @@
 	if(!victim) //likely a mapspawned one
 		victim = new(src)
 		victim.real_name = random_unique_name(prob(50))
-	spear = locate(speartype) in parts_list
+	spear = locate(/obj/item/spear) in parts_list
```

Comment:
revert. 

the only reason it's not runtiming anymore is because you're no longer passing a null value to locate(), but this isn't correct. this will never fine the original part if you construct a bonepike or bamboopike since you've defined a specific type.